plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ProgramConverter.kt

```diff
@@ -92,18 +91,21 @@ class ProgramConverter(val session: FirSession, override val config: PluginConfi
         signature: NamedFunctionSignature,
         returnTarget: ReturnTarget,
         scopeDepth: Int,
+        returnVarCtxt: ReturnVarSubstitutionContext? = null,
     ): StmtConversionContext {
+        val rootResolver =
+            RootParameterResolver(
+                this@ProgramConverter,
+                signature,
+                signature.symbol.valueParameterSymbols,
+                signature.labelName,
+                returnTarget,
+            )
         val methodCtx =
             MethodConverter(
                 this@ProgramConverter,
                 signature,
-                RootParameterResolver(
-                    this@ProgramConverter,
-                    signature,
-                    signature.symbol.valueParameterSymbols,
-                    signature.labelName,
-                    returnTarget
-                ),
+                if (returnVarCtxt == null) rootResolver else SubstitutedReturnParameterResolver(rootResolver, returnVarCtxt),
```

Comment:
Is this really worth doing here, and not building the resolver properly before the call?